src/Table/Table.tsx

```diff
@@ -521,483 +617,540 @@ interface TableProps {
+                            subRows: currentRow?.subRows.map((subRow: TableRow) => ({ ...subRow, expanded: false }))
+                        };
                     }
-                );
-            }
-            return currentRow;
-        });
+                    return currentRow;
+                });
 
-        const expandedRowList: Array<TableRow> = updatedOriginalRows.filter((item: TableRow) => {
-            return item.expanded || item.subRows.some((sub: TableRow) => sub.expanded);
-        }).map((newRow: TableRow) => {
-            return {
-                ...newRow,
-                subRows: newRow.subRows.filter((subRowItem: TableRow) => subRowItem.expanded)
-            };
-        });
+                const expandedRowList: Array<TableRow> = updatedOriginalRows
+                    .filter((item: TableRow) => {
+                        return item.expanded || item.subRows.some((sub: TableRow) => sub.expanded);
+                    })
+                    .map((newRow: TableRow) => {
+                        return {
+                            ...newRow,
+                            subRows: newRow.subRows.filter((subRowItem: TableRow) => subRowItem.expanded)
+                        };
+                    });
 
-        setCurrentTableRows(updatedRows);
-        setTableRows(updatedOriginalRows);
-        setTableRowsImage(updatedOriginalRows);
-        props.onRowExpanded(expandedRowList);
-    }, [tableRows, currentTableRows, props.onRowExpanded]);
-    // functions -----------------------------------------------------------------------------
-    /**
-     *
-     * @param rows The table or or data to initialize rows from
-     */
-    const getRows = React.useCallback((rows: Array<DataItem>): Array<TableRow> => {
-        const updatedRows: Array<TableRow> = rows.map((row: TableRow, index: number) => {
-            const updatedCells: Array<Cell> = Object.keys(row).filter((key: string) => {
-                return key !== "rowContentDetail" && key !== "subRows";
-            }).map((accessor: string): Cell => {
-                return {
-                    id: accessor,
-                    accessor,
-                    value: row[accessor]
-                };
-            });
+                setCurrentTableRows(updatedRows);
+                setTableRows(updatedOriginalRows);
+                setTableRowsImage(updatedOriginalRows);
+                props.onRowExpanded(expandedRowList);
+            },
+            [tableRows, currentTableRows, props.onRowExpanded]
+        );
+        // functions -----------------------------------------------------------------------------
+        /**
+         *
+         * @param rows The table or or data to initialize rows from
+         */
+        const getRows = React.useCallback((rows: Array<DataItem<any>>): Array<TableRow> => {
+            const updatedRows: Array<TableRow> = rows.map((row: TableRow, index: number) => {
```

Comment:
`rows?.map` else the code will break when data is set to null

Reply:
I think it makes sense here.

Reply:
Just remember you put a default value in return since `rows?.map` might return null in case there are no rows. Since the logic of the component relies on the existence of an array it might crash here.

Something like this:
```typescript
rows?.map(() => { /** your logic */ }) || []
```